google/cloud/aiplatform/models.py

```diff
@@ -2036,3 +2058,139 @@ def list(
+            ValueError if model does not support exporting.
+
+            ValueError if invalid arguments or export formats are provided.
+        """
+
+        # Model does not support exporting
+        if not self.supported_export_formats:
+            raise ValueError(f"The model `{self.resource_name}` is not exportable.")
+
+        # No destination provided
+        if not any((artifact_destination, image_destination)):
+            raise ValueError(
+                "Please provide an `artifact_destination` or `image_destination`."
+            )
+
+        export_format_id = export_format_id.lower()
+
+        # Unsupported export type
+        if export_format_id not in self.supported_export_formats:
+            raise ValueError(
+                f"'{export_format_id}' is not a supported export format for this model. "
+                f"Choose one of the following: {self.supported_export_formats}"
+            )
+
+        content_types = gca_model_compat.Model.ExportFormat.ExportableContent
+        supported_content_types = self.supported_export_formats[export_format_id]
+
+        if (
+            artifact_destination
+            and content_types.ARTIFACT not in supported_content_types
+        ):
+            raise ValueError(
+                "This model can not be exported as an artifact in '{export_format_id}' format. "
+                "Try exporting as a container image by passing the `image_destination` argument."
+            )
+
+        if image_destination and content_types.IMAGE not in supported_content_types:
+            raise ValueError(
+                "This model can not be exported as a container image in '{export_format_id}' format. "
+                "Try exporting the model artifacts by passing a `artifact_destination` argument."
+            )
+
+        # Construct request payload
+        output_config = gca_model_service_v1.ExportModelRequest.OutputConfig(
+            export_format_id=export_format_id
+        )
+
+        if artifact_destination:
+            output_config.artifact_destination = gca_io_v1.GcsDestination(
+                output_uri_prefix=artifact_destination
+            )
+
+        if image_destination:
+            output_config.image_destination = gca_io_v1.ContainerRegistryDestination(
+                output_uri=image_destination
+            )
+
+        _LOGGER.log_action_start_against_resource("Exporting", "model", self)
+
+        operation_future = self.api_client.export_model(
+            name=self.resource_name, output_config=output_config
+        )
+
+        _LOGGER.log_action_started_against_resource_with_lro(
+            "Export", "model", self.__class__, operation_future
+        )
+
+        # Block before returning
+        self._wait_on_export(operation_future=operation_future, sync=sync)
+
+        _LOGGER.log_action_completed_against_resource("model", "exported", self)
+
+        return json_format.MessageToDict(operation_future.metadata.output_info._pb)
```

Comment:
If I only provide one output target do I still get both keys in the dict and if so, what is the value for the key representing the output that wasn't set?

Reply:
The only keys in the dict are for the output targets that were set in the export model request.

```py
from google.cloud import aiplatform

m = aiplatform.Model('8842430840248991744')

m.export_model(
    export_format_id='tf-saved-model',
    artifact_destination='gs://vinnys-mbsdk-staging-uscentral1/'
)
```
returns
```py
{'artifactOutputUri': 'gs://vinnys-mbsdk-staging-uscentral1/model-8842430840248991744/tf-saved-model/2021-04-30T17:22:41.298770Z'}
```